server/src/test/java/org/elasticsearch/cluster/coordination/ElasticsearchNodeCommandTests.java

```diff
@@ -38,61 +45,101 @@
 
 public class ElasticsearchNodeCommandTests extends ESTestCase {
 
-    public void testLoadStateWithoutMissingCustoms() throws IOException {
-        runLoadStateTest(false, false);
-    }
-
     public void testLoadStateWithoutMissingCustomsButPreserved() throws IOException {
-        runLoadStateTest(false, true);
+        runLoadStateTest(false);
     }
 
     public void testLoadStateWithMissingCustomsButPreserved() throws IOException {
-        runLoadStateTest(true, true);
+        runLoadStateTest(true);
     }
 
-    public void testLoadStateWithMissingCustomsAndNotPreserved() throws IOException {
-        runLoadStateTest(true, false);
+    @Override
+    public Settings buildEnvSettings(Settings settings) {
+        // we control the data path in the tests, so we don't need to set it here
+        return settings;
     }
 
-    private void runLoadStateTest(boolean hasMissingCustoms, boolean preserveUnknownCustoms) throws IOException {
-        final Metadata latestMetadata = randomMeta();
-        final XContentBuilder builder = JsonXContent.contentBuilder();
-        builder.startObject();
-        Metadata.FORMAT.toXContent(builder, latestMetadata);
-        builder.endObject();
-
-        XContentParserConfiguration parserConfig = hasMissingCustoms
-            ? parserConfig().withRegistry(ElasticsearchNodeCommand.namedXContentRegistry)
-            : parserConfig();
-        Metadata loadedMetadata;
-        try (XContentParser parser = createParser(parserConfig, JsonXContent.jsonXContent, BytesReference.bytes(builder))) {
-            loadedMetadata = Metadata.fromXContent(parser);
-        }
```

Comment:
This round-trip does not seem to directly test `ElasticsearchNodeCommand` at all. It is replaced directly calling `ElasticsearchNodeCommand#loadTermAndClusterState`.